### PR TITLE
Add image info dialog (#417)

### LINKED ISSUE #417: view file information or image header
We should be able to view the image header of the current frame.  One way to do this is to add  "image header" or "image info"  sub menu to the "view" menu.  This would bring up a text widget containing the image header. 

Alternately one could add an info panel that contains the file information.  Not sure which is more useful the file information or the header.  The header is more complete, but the file info is more readable.

### PR DESCRIPTION
**Pull request by TienHao**
closes #417 

* Add image info dialog component

* Add image info dialog css

* Add image info store

* Add "Image Info" in submenu "View"

### CODE REVIEW

**Comment by duidae** [comment]
1. issue - when appending multiple frames, the image info should show the info of the active frame
2. instead of saving the lastest in the imageinfo store, i think its better to wrap the image info into frame store as a subsidiary attribute, so that it will be easier to maintain when adding/deleting a frame
3. btw, @veggiesaurus  @kswang1029  will it be better to also have an info button in the tool bar?
image

**Comment by veggiesaurus** [comment]
We already have the contents of the image info in frame (in the extended file info). No need to duplicate it, just use that instead.

@duidae I'd prefer not to add it to the toolbar, it's already getting quite long. I think we should have a separate "dialogs" toolbar at the top near the widgets. We can have buttons for preferences, file info, overlay config, contour config and so on.

File: src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx [hunk 1]
```diff
@@ -30,6 +30,10 @@ export class FileBrowserDialogComponent extends React.Component<{ appStore: AppS
         }
 
         if (fileBrowserStore.browserMode === BrowserMode.File) {
+            const imageInfoStore = this.props.appStore.imageInfoStore;
```
Comment (veggiesaurus) [inline src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx]: reuse `appStore.activeFrame.frameInfo.fileInfoExtended` instead, no need to copy it here

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+@observer
+export class ImageInfoDialogComponent extends React.Component<{ appStore: AppStore }> {
+
+    private handleTabChange = (newId: TabId) => {
+        this.props.appStore.imageInfoStore.setSelectedTab(newId);
+    };
+
+    private renderInfoPanel = () => {
+        const imageInfoStore = this.props.appStore.imageInfoStore;
```
Comment (veggiesaurus) [inline src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.tsx]: reuse `appStore.activeFrame.frameInfo.fileInfoExtended` for these values

**Review by veggiesaurus — CHANGES_REQUESTED** [review]
mostly looks ok, but we should reuse information in `FrameStore` instead of duplicating. This will also fix the bug @duidae found.

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        .bp3-input-group {
+            margin: 10px 5px 5px;
+        }
+
+        .image-info-panes {
+            display: flex;
+            height: 100%;
+            
+            .image-info-pane {
+                width: 95%;
```
Comment (veggiesaurus) [inline src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss]: I don't think these margins are correct. Rather just use a (much smaller) fixed margin and use CSS `calc` to work out the width. I'm not sure why we need a 40px `margin-bottom`.

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    height: 100%;
+    user-select: none;
+
+    .bp3-dialog-body {
+        height: 100%;
+        overflow: hidden;
+        margin-top: 10px;
+        margin-left: 25px;
+
+        .bp3-input-group {
```
Comment (veggiesaurus) [inline src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss]: is this used?

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss [hunk 5]
```diff
@@ -0,0 +1,38 @@
+.image-info-dialog {
+    min-width: 60%;
+    height: 100%;
+    user-select: none;
+
+    .bp3-dialog-body {
+        height: 100%;
+        overflow: hidden;
+        margin-top: 10px;
+        margin-left: 25px;
```
Comment (veggiesaurus) [inline src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss]: don't think we need such a large left margin. In any case, it should be symmetric

File: src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx (hunk 1 above)
Comment (TienHao) [reply src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx]: OK, I will remove ImageInfoStore. It is much simple to implement image info dialog with FrameStore.

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss (hunk 5 above)
Comment (TienHao) [reply src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss]: This is a legacy from FileBrowserComponent. I will adjust it.

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss (hunk 4 above)
Comment (TienHao) [reply src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss]: No, it is not used. This is also a legacy from FileBrowserComponent. I will remove it.

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.tsx (hunk 2 above)
Comment (TienHao) [reply src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.tsx]: I will remove all ImageInfoStore code.

File: src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss (hunk 3 above)
Comment (TienHao) [reply src/components/Dialogs/ImageInfo/ImageInfoDialogComponent.scss]: The 40px margin-bottom is for fully seeing the last line of headers information.

File: src/components/Dialogs/FileInfoDialog/FileInfoDialogComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import {DraggableDialogComponent} from "components/Dialogs";
+import {FileInfoComponent, InfoType} from "components/FileInfo/FileInfoComponent";
+import {CARTA} from "carta-protobuf";
+import {AppStore} from "stores";
+import "./FileInfoDialogComponent.css";
+
+@observer
+export class FileInfoDialogComponent extends React.Component<{ appStore: AppStore }> {
+
+    @observable fileInfoExtended: CARTA.FileInfoExtended;
```
Comment (duidae) [inline src/components/Dialogs/FileInfoDialog/FileInfoDialogComponent.tsx]: unused variable?

**Review by duidae — APPROVED** [review]

File: src/App.tsx [hunk 7]
```diff
@@ -132,6 +133,7 @@ export class App extends React.Component<{ appStore: AppStore }> {
                         {this.renderHotkeys()}
                     </div>
                 </Dialog>
+                <FileInfoDialogComponent appStore={appStore}/>
```
Comment (veggiesaurus) [inline src/App.tsx]: I suggest you move this up where the rest of the simple dialogs are

**Review by veggiesaurus — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 7 above)
Comment (TienHao) [reply src/App.tsx]: OK

**Review by veggiesaurus — APPROVED** [review]

**Pull request merged**